### DISCUSSION TITLE
Sugestions.

### DISCUSSION THREAD

**Discussion by Bernardo-BR** [sircharlo/mmm-refactor#271] [Ideas]
Firstly, thank you very much for your efforts in creating such a useful application. We brothers from Brazil are very grateful. Jehovah continue to bless you for all your dedication to this project !

The first suggestion is to have an option in the program, for obs scenes, where they are always visible, like in the old M3. Which makes actions in meetings much faster and with no errors.

![imagem_2024-10-10_215321294](https://github.com/user-attachments/assets/f051a2a9-be23-4cf2-9984-a8a082998e74)

My second suggestion is a option to create personalized tab's in the section "meeting media". For example, a visit of circuit overseer, or a public speech. Which will make it easier to use, and more organized as well.

![image](https://github.com/user-attachments/assets/ab21a030-fc60-49d6-9e62-cda66865d1c4)

**Comment by sircharlo** [comment]
> Firstly, thank you very much for your efforts in creating such a useful application. We brothers from Brazil are very grateful. Jehovah continue to bless you for all your dedication to this project !

Thank you for your kind words!

---

> The first suggestion is to have an option in the program, for obs scenes, where they are always visible, like in the old M3. Which makes actions in meetings much faster and with no errors.

@ux-git Do you think you could mock up what a "docked" OBS scene control area could look like? 
If you can think of a clean way to have OBS controls docked and always visible, I could implement it and add a toggle to the OBS integration settings to choose either the current view (my personal favorite) or a docked view where the scene buttons are always visible (which might be useful in some congregations that make heavy use of scene switching).

---

> My second suggestion is a option to create personalized tab's in the section "meeting media". For example, a visit of circuit overseer, or a public speech. Which will make it easier to use, and more organized as well.

May I ask why doing this would be better than the current method? Keep in mind, the goal with this app is simplicity.
As it is right now, if a user adds custom media, it is added to the Imported Media section up top. The user can drag/drop that media into any other section to organize it as desired, and the imported media will become a part of that section.
I struggle to see why a meeting would need more sections. Right now, the layout is:

- **Midweek meeting:** the 3 usual sections (TFGW, LAC, AYFM), plus the Imported Media section if user adds custom media. The custom media can be sorted and moved to any of the 3 other sections as needed.
- **Weekend meeting:** the 1 usual section (WT), plus the Imported Media section if user adds custom media. The user is prompted to add an opening song if none was added. Any additional custom media that is added for the public talk will be in the first section (Imported Media) and can be sorted as needed.

The only scenario where an additional section could be useful from what I see is for the CO week, where the weekend meeting has 3 sections instead of 2. An argument could also maybe be made that the midweek meeting goes from 3 sections to 4 during the week of the CO.

Are there other scenarios that I'm not aware of?

**Reply by Bernardo-BR** [reply]
> May I ask why doing this would be better ...



@sircharlo Hello, thank you for your time in responding.

The suggestion **2** is really for special weeks, like the visit of the CO. The settings for regular meetings are already exceptional, I see no reason to change them as well!

As for the scenarios, I believe that in public meetings, there aren't scenarios significant enough to require this function. How do you say, are the section of imported media in program.

Also I'm sorry for the awkward English, Google Translate is horrible 😂.

**Reply by e-Feitosa** [reply]
[MSG em pt-BR]
@Bernardo-BR, o irmão configurou essa opção para reuniões especiais?
Tanto no M3 atual, quanto na versão nova, vai ter a opção para eventos especiais.
Não sei bem se era isso que estava tentando explicar, qualquer coisa me chama para falarmos melhor [🗣️](https://wa.me/message/RQ4MGK6LWJ2XC1)

**Comment by ux-git** [comment]
`Do you think you could mock up what a "docked" OBS scene control area could look like?
`
@sircharlo, I'll take a look together with Stas today/tomorrow. We already have the OBS button in the action island, so let's see if we can fully utilize it.

**Comment by e-Feitosa** [comment]
@sircharlo, a hint for the scenes of OBS Studio, when activated (in use), the scenes would appear in the sidebar, that is, up to 6 scenes would appear from 1 to 6 to quickly change scene and for more scenes would hide the rest if you had more than 6 scenes. That is, I would take better advantage of the sidebar of M3
![image](https://github.com/user-attachments/assets/ac764412-9564-4081-b0f9-eb854bf5aa59)

**Reply by ux-git** [reply]
@sircharlo what's your opinion on using the sidebar in a such way? Maybe even at the bottom to separate quick controls from the main navigation.

**Reply by sircharlo** [reply]
I thought maybe the action island could get a second row. When the "always show obs scene buttons" settings toggle is enabled, the scene buttons would be displayed in that second row instead of being hidden behind the OBS button. When disabled, the second row is hidden and the OBS button is shown, like it is at the present. 

I guess I'd like to keep sections consistent in their purpose. 

Here's how I see it:

- Side = app navigation
- Bottom = quick actions that are done during the meeting
- Top = advanced / disruptive actions that are done before the meeting

**Reply by e-Feitosa** [reply]
My **suggestion** would be something like this, when activated the option OBS Studio, a bar below the main controls, would be visible with the scenes of OBS Studio in case you use multiple scenes.
![image](https://github.com/user-attachments/assets/d7f57509-19ed-418b-96c4-a2d87d318ce8)

**Reply by ux-git** [reply]
@e-Feitosa, @Bernardo-BR have you tried using OBS hotkeys for switching OBS scenes let's say with numpad buttons? If not yet, I recommend you to give it a try. From my experience, it's way faster than any on-screen controls, simply because buttons are bigger, tactile and always available in the same spot, even if you open another app in the foreground (for example, Zoom).

**Reply by e-Feitosa** [reply]
@ux-git , is interesting, but we do not use here, as simple as it seems, using shortcut was more complicated than clicking on the scene to change the camera. Many do not get, end up forgetting the combination and then when they are going to change the scene, is already elsewhere, clicking on the scene to change, has been more practical for several operators, not only here but in other congregations that asked for help in installation.

**Comment by e-Feitosa** [comment]
Very good the new adjustments for the week of the visit! Thanks for the excellent work of the M3 team
![image](https://github.com/user-attachments/assets/91d75ef2-8109-4820-93bd-4871ec7dcdce)
